snuba/query/processors/logical/custom_function.py

```diff
@@ -133,3 +143,37 @@ def apply_function(expression: Expression) -> Expression:
                 return expression
 
         query.transform_expressions(apply_function)
+
+
+class ApdexProcessor(LogicalQueryProcessor):
+    def __init__(self) -> None:
+        self.__processor = _CustomFunction(
+            "apdex",
+            [("column", ColType({UInt})), ("satisfied", LiteralType({int}))],
+            simple_function(
+                "divide(plus(countIf(lessOrEquals(column, satisfied)), divide(countIf(and(greater(column, satisfied), lessOrEquals(column, multiply(satisfied, 4)))), 2)), count())"
+            ),
+        )
+
+    def process_query(self, query: Query, settings: QuerySettings) -> None:
```

Comment:
I think `process_query` can still be defined in the parent class.

Reply:
There's no parent class here. It's a wrapper class around the _CustomFunction processor